src/reactive-membrane.ts

```diff
@@ -35,15 +35,20 @@ export interface ObservableMembraneInit {
     valueObserved?: ReactiveMembraneAccessCallback;
     valueDistortion?: ReactiveMembraneDistortionCallback;
     valueIsObservable?: ReactiveMembraneObservableCallback;
+    magicPropertyKey?: PropertyKey;
 }
 
-function createShadowTarget(value: any): ReactiveMembraneShadowTarget {
+function createShadowTarget(value: any, magicPropertyKey: PropertyKey | undefined): ReactiveMembraneShadowTarget {
     let shadowTarget: ReactiveMembraneShadowTarget | undefined = undefined;
     if (isArray(value)) {
         shadowTarget = [];
     } else if (isObject(value)) {
         shadowTarget = {};
     }
+    if (!isUndefined(magicPropertyKey)) {
+        // installing the magic property as configurable, non-writable and non-enumerable
+        ObjectDefineProperty(shadowTarget, magicPropertyKey, { configurable: true });
```

Comment:
configurable, non-writable is the optimal definition since it can still change if needed, and the object invariants will not apply

Reply:
When would we reuse the same shadow target and reset the property?

Reply:
we will never reuse a shadow target, is that what you're asking?

Reply:
Yes. I don't understand why the `magicPropertyKey` is set to configurable.